### DISCUSSION TITLE
LUA: calculate length (m) and area (m^2)

### DISCUSSION THREAD

**Discussion by tordans** [osm2pgsql-dev/osm2pgsql#1756] [Ideas]
Would it be possible to calculate …
- the length (way) and – _update: https://github.com/openstreetmap/osm2pgsql/issues/1757_
- the area (area/polygons) – _update: done already_ 

… of objects in LUA? If so, this would be a convenient addition to the new [LUA geometry helper](https://osm2pgsql.org/doc/manual.html#geometry-objects-in-lua).

---

With this I could filter for example small ponds (that lack specific tags) or small segments of road that I am not interested in. (Those filter are a bit rough, since I only look at one object, of course, but good enough for some use cases.)

**Comment by joto** [comment]
There is no way to calculate length currently, but there is an `area()` function that does what you want and documented in the chapter you are linking to.

**Reply by joto** [reply]
I have opened an issue #1757 for the length() function.

**Reply by joto** [reply]
Note that the `area()` function (and also the `length()` function when it appears) will report their results in map units (degrees for lon/lat, etc.) This is for consistency with those function in other systems and also because it is cheaper to calculate. How useful this is compared to functions reporting in meters or square meters depends on the use case. We might want to have variants for that, too.

**Reply by tordans** [reply]
>ill report their results in map units (degrees for lon/lat, etc.) 

I wanted to ask for the unit because the docs don't talk about them. 

Should I make a PR to add this info to the docs? (Something [along the lines of](https://osm2pgsql.org/doc/manual.html#special-tags) "(calculated in the units of the projection, normally Mercator meters)")

This might be a noob question, but just to be sure: This does mean that for mercator/3857 area and length will return a float in meter, right?

**Reply by joto** [reply]
PR: Please do. 

mercator/3857: No, not in meter, but something close enough in lower latitudes and for shorter distances. Read https://en.wikipedia.org/wiki/Mercator_projection for all the details. :-)
